### ISSUE TITLE
[META] Apply Style Guide to Backdrop Admin Theme/System CSS

### ISSUE THREAD

**Issue by wesruv** [backdrop/backdrop-issues#736]
We have a styleguide at: http://backdrop-ops.github.io/styleguide  

This meta-ticket lists all of the tasks related to making our administration theme match the styleguide.(Feel free to grab any ticket you can tackle)

##### TODO for 1.1.0
- [x] Element Styles - https://github.com/backdrop/backdrop-issues/issues/738
- [x] Form elements (Pt 1) - https://github.com/backdrop/backdrop-issues/issues/753
- [x] Form elements (Pt 2) - https://github.com/backdrop/backdrop-issues/issues/754
- [x] Drop Button - https://github.com/backdrop/backdrop-issues/issues/785
- [x] Marks for New and updated - https://github.com/backdrop/backdrop-issues/issues/760
- [x] Installer font needs fixing - https://github.com/backdrop/backdrop-issues/issues/879
- [x] Fonts - https://github.com/backdrop/backdrop-issues/issues/875
- [x] Fix Italics in font - https://github.com/backdrop/backdrop-issues/issues/878
- [x] Status messages - https://github.com/backdrop/backdrop-issues/issues/739 
- [x] Breadcrumbs - https://github.com/backdrop/backdrop-issues/issues/759
- [x] Tabs - https://github.com/backdrop/backdrop-issues/issues/737
- [x] Tables - https://github.com/backdrop/backdrop-issues/issues/740 
- [x] Vertical Tabs - https://github.com/backdrop/backdrop-issues/issues/756
- [x] Pager - https://github.com/backdrop/backdrop-issues/issues/784
- [x] Progress Bar - https://github.com/backdrop/backdrop-issues/issues/758 (wesruv)

##### Followups based on completed issues above:
- [x] Table style cleanup - https://github.com/backdrop/backdrop-issues/issues/874
- [x] Fix up Layout page #891 (wesruv)
- [x] Form styles and admin tables #888 (graham72)
- [x] Status messages cleanup #880 (wesruv)
- [x] Enabling modules should not use "button-danger" class #912

##### Things that can go in after 1.1.0
- [x] Recent log messages table - https://github.com/backdrop/backdrop-issues/issues/876 (wesruv)
- [x] New style primary tabs have a 1px gap in Firefox - https://github.com/backdrop/backdrop-issues/issues/877
- [x] Fancy Buttons - https://github.com/backdrop/backdrop-issues/issues/883
- [x] Fieldsets - https://github.com/backdrop/backdrop-issues/issues/755 (wesruv)

(@quicksketch will check off items from this list as they get merged into the 1.x-new-style branch.)

##### Blocked issues  (held up by backdrop-ops/styleguide#11)
- [ ] **BLOCKED** [UX] Button focus - https://github.com/backdrop/backdrop-issues/issues/909

### Implementation Guidelines
We want to avoid making changes that will break current Backdrop sites, try to avoid markup and style changes in the system CSS if they'll break people's themes. Changes that may cause this kind of breakage can be added to the admin theme (Seven) instead of System css.

Check the Styleguide feedback ticket to see if there are any issues for a ticket you may grab. https://github.com/backdrop-ops/styleguide/issues/11

Anyone who wants to work on implementation of the Backdrop admin theme [styleguide](http://backdrop-ops.github.io/styleguide) should start by installing backdrop + and adding the [styleguide module](https://github.com/backdrop-contrib/styleguide). That will give you the correct CSS + JS as a starting point, including working dialogs!

Screenshots in tickets may not be up to date, consider the style guide the canonical source.

**Comment by docwilmot** [comment]
I hadnt looked at this for a while. This is freakin fantastic. Well done.

**Comment by quicksketch** [comment]
@docwilmot Darius is still doing another round of changes on the Styleguide as well. If you'd like to add any additional input we're tracking discussion on this iteration at https://github.com/backdrop-ops/styleguide/issues/11.

Agreed though, this looks amazing. I can't wait to see it in core.

**Comment by jenlampton** [comment]
Anyone who wants to work on implementation of the Backdrop admin theme [styleguide](http://backdrop-ops.github.io/styleguide) should start by installing backdrop + and adding the [styleguide module](https://github.com/backdrop-contrib/styleguide). That will give you the correct CSS + JS as a starting point, including working dialogs!

**Comment by wesruv** [comment]
@ronnienorwood, moved the work that https://github.com/backdrop/backdrop-issues/issues/201#issuecomment-73471691 was going to cover into this ticket. Feel free to grab anything that's not blocked. I grabbed Tabs and Status messages, hoping to have some work done on those soon :smile:

**Comment by ronnienorwood** [comment]
Cool @wesruv.  Thanks for the update.  Hope to get some work done on some of these too.

**Comment by jenlampton** [comment]
I grabbed tables & pager.  Both have PRs.

@ronnienorwood the most important one is the most general element styles for seven - since that will have the new font. I think you did most of that work already, maybe it would be quick for you to repeat those changes? I made some updates to the styleguide to unblock the remaining issues there if you want to take a stab at that one.

**Comment by ronnienorwood** [comment]
@jenlampton  I assigned #738 to myself.  I will work on that today.  What is the best way to test changes ??  Does the styleguide module create a styleguide page that you can use for testing ??  When I tested it before, I just used various admin pages to check changes.

**Comment by wesruv** [comment]
@ronnienorwood it'll create a page per theme. I keep Seven, Bartik, and a Bare theme (that has no tpl, css, anything) open and test them in different tabs. Only reason to have bartik open is that if you mess up that theme you'll likely mess with other user's custom themes.

When you have styleguide installed you can go to: 
`/admin/appearance/styleguide/bartik`
`/admin/appearance/styleguide/seven`
or whatever theme.

**Comment by ronnienorwood** [comment]
Thanks @wesruv for the clarification.

**Comment by quicksketch** [comment]
These are all looking pretty good. Tonight I merged together locally all the issues we had PRs for. They all look good but together it makes for a pretty mish-mashed UI.

To facilitate getting all these things working together, I'd like to propose we use a single feature branch for all the changes to be combined together. That will make styling the appearance of them much easier to see all at once, while we can continue reviewing the PRs a-la-carte as they come in. If I created a branch for `1.x-new-style` for example and merged in all the current PRs, would that help bring everything together?

**Comment by wesruv** [comment]
@quicksketch I like it

**Comment by quicksketch** [comment]
Hi everyone, I just wanted to say excellent job on moving this forward. We have about 2 weeks until the 1.1.0 release, and I think we've gotten this to a point where it's very-nearly acceptable. Let's continue pounding through the remaining issues and look for a merge into 1.x next week. Some remaining tasks (e.g. the progress bar) may not reach completion by 1.1.0, but that's acceptable for us to delay until 1.2.0 (Aug 15, 2015). Any regressions or UX problems we've created can of course be addressed in bugfix releases in 1.1.x. Keep up the great work!

**Comment by quicksketch** [comment]
With 1 week remaining until release, I think it's time to merge in this new branch into the main 1.x branch. The TODO list above is looking great, with only progress bars in the 1.1.0 target. I'd love to get this into 1.x to get maximum exposure before our release. Do you guys think we're ready? @wesruv, @Graham-72, @jenlampton, et al?

**Comment by wesruv** [comment]
I feel good about it

**Comment by docwilmot** [comment]
Looks good. Go for it.

**Comment by quicksketch** [comment]
We are merged!! https://github.com/backdrop/backdrop/commit/391131b2d27b62b9567925c73e23117b8e1c97a5

Great job everyone! Let's continue to finish up the lingering issues we know of and also file new issues as we come across them.

**Comment by Graham-72** [comment]
That is really good news to wake up to here in the UK. Much better (for some) than our election results yesterday morning. My brain is still a bit addled from all the recent news/propaganda/misinformation of recent weeks. How/where do we record the lingering and new issues? I added one yesterday to https://github.com/backdrop/backdrop-issues/issues/888 but maybe this was the wrong place?

**Comment by docwilmot** [comment]
Brilliant, looks fantastic!
Issues: I upgraded by simply copy/pasting the "core" folder. Didnt have pathauto or token contrib installed. 

**Issue 1:**
But first thing I did, for no reason was flush all caches, and I got this:
`PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'bd.cache_token' doesn't exist: TRUNCATE {cache_token} ; Array ( ) in backdrop_flush_all_caches() (line 7981 of C:\wamp\www\backdrop\backdrop\core\includes\common.inc).`

I hit the back button, ran upgrade and all was well.

**Issue 2:**
the default body field for add content is huge!! Its set at 20 rows, could we make it 10? Unless youre writing a PhD thesis in there, I think 10 would do. *Fullscreen* shots follow:

See 20 rows(current):
![rows20](https://cloud.githubusercontent.com/assets/1042283/7550154/e8365f6a-f61c-11e4-9e54-ced83cd5a73d.PNG)

See 10 rows:
![rows10](https://cloud.githubusercontent.com/assets/1042283/7550158/fbaf01a0-f61c-11e4-9882-161ef265e0b7.PNG)

D8 for comparison  (9 rows):
![d8](https://cloud.githubusercontent.com/assets/1042283/7550161/0fbeb604-f61d-11e4-87b5-d4d8b4c878c9.PNG)

**Issue 3:**
On node add page the image field is a bit disjointed looking. Could it wrap in a fieldset or something? (this isnt an issue with the style guide as such, but I think the new theme makes it more obvious)

![capture](https://cloud.githubusercontent.com/assets/1042283/7550176/8a58538e-f61d-11e4-93cd-28e5c7b9d660.PNG)

Thats it, otherwise happy.

**Comment by docwilmot** [comment]
One more: the hover color for table rows is a bit too yellow.

**Comment by serundeputy** [comment]
this looks great!  thanks for all the hard work; so cool.

**Comment by docwilmot** [comment]
I got no comment on my https://github.com/backdrop/backdrop-issues/issues/736#issuecomment-100474068. But the textarea is still too big.

**Comment by quicksketch** [comment]
> I got no comment on my https://github.com/backdrop/backdrop-issues/issues/736#issuecomment-100474068. But the textarea is still too big. 

I made a dedicated issue at https://github.com/backdrop/backdrop-issues/issues/1034. This is more related to text.module's defaults than the styleguide (I think). So let's solve that as a stand-alone issue.

**Comment by jenlampton** [comment]
yay!

**Comment by klonos** [comment]
Yes yay! By the way Jen, I love your new avatar ...you look amazing there.

**Comment by jenlampton** [comment]
aw, thanks @klonos. It was my wedding day ;)
